lib/ansible/modules/database/mongodb/mongodb_replicaset.py

```diff
@@ -0,0 +1,412 @@
+#!/usr/bin/python
+
+# (c) 2018, Rhys Campbell <rhys.james.campbell@googlemail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: mongodb_replicaset
+short_description: Initialises a MongoDB replicaset before authentication has been turned on and then validates the configuration when it has been turned on.
+description:
+    - Initialises a MongoDB replicaset before authentication has been turned on and then validates the configuration when it has been turned on.
+    - Validation confirms the replicaset set name only.
+    - This may change in the future to involve adding and removing members.
+author: Rhys Campbell (@rhysmeister)
+version_added: "2.8"
+options:
+    login_user:
+        description:
+            - The username used to authenticate with.
+        required: false
+    login_password:
+        description:
+            - The password used to authenticate with.
+        required: false
+    login_database:
+        description:
+            - The database where login credentials are stored.
+        required: false
+        default: "admin"
+    login_host:
+        description:
+            - The MongoDB hostname to login to.
+        required: false
+        default: "localhost"
+    login_port:
+        description:
+            - The MongoDB port to login to.
+        required: false
+        default: 27017
+    replica_set:
+        description:
+            - Replicaset name.
+        required: false
+        default: rs0
+    members:
+        description:
+            - A comma-separated consisting of the replicaset members.
+            - Supply as a simple csv string, i.e. mongodb1:27017,mongodb2:27017,mongodb3:27017.
```

Comment:
Can this alternatively also be a YAML list? This is common in other modules (the package manager modules come to mind) and I'd like to see it carried over here if possible.

Reply:
I've made some changes to allow the replicaset members to be supplied as a list and update the examples in the documentation section.

Got into a bit of a tiz with a rebase. Not sure why at the moment but it's cause the shippable tests to be cancelled.